argo/providers/vai/provider.go

```diff
@@ -193,8 +202,40 @@ func (vaip VAIProvider) DeletePipeline(ctx context.Context, providerConfig VAIPr
 	return nil
 }
 
+func ExtractFromMap[T any](targetMap map[string]any, fieldName string) (T, error) {
+	result, ok := targetMap[fieldName].(T)
+	if !ok {
+		err := errors.New("Failed extracting field " + fieldName + "from a given map")
+		var emptyResult T
+		return emptyResult, err
+	}
+	return result, nil
+}
```

Comment:
think this can be simplified, as I understand it from reading `.(T)` is a type assertion and will assign the value to `result` if the type matches, otherwise `result` will be empty. So by naming the return values, you can just assign the error to `err` if `ok` is false (bcos result will be empty anyway)

saves you having to construct an empty value for `T`

```suggestion
func ExtractFromMap[T any](targetMap map[string]any, fieldName string) (result T, err error) {
	result, ok := targetMap[fieldName].(T)
	if !ok {
		err = fmt.Errorf("failed extracting field %s from the given map", fieldName)
	}
	return result, err
}
```

Reply:
Cool, yes, that works, have combined with the error message change